dask/dataframe/io/io.py

```diff
@@ -613,6 +617,98 @@ def from_delayed(
     return df
 
 
+@insert_meta_param_description
+def from_callable(
+    func,
+    npartitions,
+    *args,
+    meta=None,
+    divisions=None,
+    prefix="from-function",
+    io_prefix="blockwise-io",
+):
+    """Create Dask DataFrame from a data-creation function.
+
+    A function (specified by the `func` argument) will be used to generate the
+    partitions of a new DataFrame collection.
+
+    Parameters
+    ----------
+    dfs : Callable
+        A function that accepts a partition index (and any other args and/or kwargs),
+        and returns a DataFrame object.
+    npartitions : int
+        The number of partitions in the resulting DataFrame collection.
+    $META
+    divisions : tuple, str, optional
+        Partition boundaries along the index.
+        For tuple, see https://docs.dask.org/en/latest/dataframe-design.html#partitions
+        For string 'sorted' will compute the delayed values to find index
+        values.  Assumes that the indexes are mutually sorted.
+        If None, then won't use index information
+    prefix : str, optional
+        Prefix to prepend to the collection key names.
+    io_prefix : str, optional
+        Prefix to prepend to blockwise "IO" tasks.
+    """
+    if not callable(func):
+        raise TypeError("Expected callable object, got %s" % type(func).__name__)
+
+    if not isinstance(npartitions, int):
+        raise TypeError("Expected integer, got %s" % type(npartitions).__name__)
+
+    # Create metadata
+    if meta is None:
+        meta = func(0, *args)
+    meta = make_meta(meta)
+
+    # Define key names
+    arg_token = tokenize(*args)
+    output_name = prefix + "-" + arg_token
+    io_name = io_prefix + "-" + arg_token
+
+    # Blockwise-transform graph is a "no-op"
+    dsk_tf = {output_name: (_noop, "_0")}
```

Comment:
It seems that a "no-op" is currently required for `optimize_blockwise` to work properly.  This ultimatly adds an extra task for each partition/chunk of the collection.  So it would be ideal to find a way to avoid this.